### PR TITLE
Refactor to clean up code, better tests, add transaction ID support

### PR DESCRIPTION
**Pull request by jraede**
This refactor does a few things:

1. **Clean up the code**. It was hard to follow and very opinionated.
2. **Add support for multiple loggers**. This allows us to create a logger per transaction that automatically adds transaction ID
3. **Sets up JSON logging for everything**. This allows logs to be parsed by logstash, etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,37 +9,68 @@ logstar
...
+#### `infof(format, ...args)`
+#### `warn(...args)`
+#### `warnf(format, ...args)`
+#### `error(...args)`
+#### `errorf(format, ...args)`
+#### `fatal(...args)`
+#### `fatalf(format, ...args)`
+
+### `transactionLogger`
+Get a logger that adds a `transaction_id` property to the meta. Useful for tracking related requests.
```
Comment (taddison92) [inline README.md]: Should this be the same logger as the other logger? By that I mean: one logger that generates a transaction ID if it needs to?

File: lib/Logger.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+  }
+
+  logf(level, format, ...args) {
+    this.log(level, util.format(format, ...args));
+  }
+}
+
+const methods = ['debug', 'info', 'warn', 'error', 'fatal'];
+methods.forEach(level => {
```
Comment (taddison92) [inline lib/Logger.js]: This is a sweet way to dynamically define these functions.

File: lib/Logger.tests.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          token: 'bar',
+          tags: 'baz,bing',
+        },
+      });
+      spyOn(logger.winston.transports.console, 'log').andReturn(true);
+      spyOn(logger.winston.transports.loggly, 'log').andReturn(true);
+      logger.log('info', 'foo');
+      expect(logger.winston.transports.console.log).toHaveBeenCalled();
+      const args = logger.winston.transports.console.log.calls[0].arguments;
+      expect(args[0]).toEqual('info');
```
Comment (taddison92) [inline lib/Logger.tests.js]: It might be overkill but having tests iterating over the different methods we care about ['info', 'debug', 'fatal'....] could be useful to guarantee that these methods all work.

**Comment by taddison92** [comment]
Looks good. I think having the transactionLogger be separate from the regular logger is less preferable than adding to logger, the ability to take an optional parameter.

File: README.md (hunk 1 above)
Comment (jraede) [inline README.md]: Yeah, but then we run into race conditions if we have code executing in parallel. Imagine this situation:

1. Execute concurrent transactions using `Promise.all`
2. In one, set the transaction ID
3. If it's a shared instance, then the other chain would get that transaction ID, or overwrite it when it tries to set it for itself.

Comment (jraede) [inline README.md]: The other option is providing the transaction ID to every call to `log`, `info`, `debug`, etc, but that would be tedious.

File: lib/Logger.tests.js (hunk 3 above)
Comment (jraede) [inline lib/Logger.tests.js]: Good point

File: README.md (hunk 1 above)
Comment (taddison92) [inline README.md]: hmm I see what you mean, sort of. It would be nice to have our services such that on original invocation of the service, set up the logger with the transaction id as a synchronous process, then later whatever operations it does use that logger that has the transaction id already set up (either created or received via parameter).

Comment (jraede) [inline README.md]: Yeah exactly. I'd like to instantiate a logger using `getTransactionLogger` on each request and wrap the request in an object that can do like `this.log`. So this is to help us get towards that.
